### ISSUE TITLE
class_toggles last active scene

### ISSUE THREAD

**Issue by Spica2** [janpaepke/ScrollMagic#345]
I use the class_toggles.html example with scene duration=0. I only want to get to the last active scene's css. But all passed scenes are active. Is there a way to set an end ancor respecting the varying containers height? I also tried $('.active').last().addClass('last'); but it seems to get overruled.

**Comment by janpaepke** [comment]
I'm not sure I understand your question.

Scenes with `duration=0` by default have no end and thus stay active indefinitely.
So if you want to have a class attached to an element for only a certain amount of scroll progress you will need to give the scene a duration.

If this doesn't help you please provide a demo (fiddle or codepen) so we can better understand what's happening and what you're trying to achieve.

**Comment by Spica2** [comment]
Ok. I'll try to explain better (a demo would not show what I want yet). In your demo, the duration is set in the script an is adusted to the content containers height manualy. If I change a content containers height by adding e.g. lines, the duration has to get adjusted manually also. I 'd like to have the same menu effect, but without fixed end positions. Therefore I see to concepts:
1. set the duration to 0 making the scene infinite. This highlights e.g. in scene 3 1,2,3 instead of only 3. adding a .active.last on the last active scene (in this example on 3) was my first attempt, without sucess.
2. adjust the duration by script to the containers height.
So, what would you recommend? Or is it already possible but I did not find it?

**Comment by janpaepke** [comment]
Yes, that's already possible: http://scrollmagic.io/examples/basic/responsive_duration.html

hope this helps?

**Comment by janpaepke** [comment]
did this solve your issue?

**Comment by janpaepke** [comment]
Sorry for the very late reply.

In v2 with `duration: 0`, scenes triggered and stayed active once passed -- there was no built-in "only the current scene is active" mode. The typical workaround was to give scenes a calculated duration based on the next trigger element's offset, effectively creating exclusive active ranges.

v3 fires `enter` and `leave` events based on actual viewport visibility, so an element that has scrolled out of view will receive a `leave` event. You can use these to toggle classes so only the currently visible section is marked active.

---
**ScrollMagic v3 is now available** -- a complete rewrite with an entirely new architecture and API. v2 will not receive further updates (2.0.9 is the final release).

👉 `npm install scrollmagic@next` -- [Docs](https://github.com/janpaepke/ScrollMagic/blob/main/README.md) · [Release notes](https://github.com/janpaepke/ScrollMagic/releases/tag/v3.0.0-beta.0) · [npm](https://www.npmjs.com/package/scrollmagic/v/3.0.0-beta.0)
